app/assets/javascripts/controllers/current_deploys_badge_ctrl.js

```diff
@@ -1,33 +1,30 @@
 samson.controller('currentDeployBadgeCtrl', function($scope, $http, SseFactory) {
   'use strict';
 
-  $scope.currentActiveDeploys = 0;
+  var $badge = $('#currentDeploysBadge');
+  $scope.currentActiveDeploys = parseInt($badge.data('count'), 10);
 
-  SseFactory.on('deploys', function(msg) {
-    if (msg.type === 'new') {
-      $scope.currentActiveDeploys += 1;
-    } else if (msg.type === 'finish') {
-      $scope.currentActiveDeploys -= 1;
-    }
-    updateBadge();
-  });
+  // connect to stream for updates once a user has the page open for a while
+  setTimeout(function(){
+    SseFactory.on('deploys', function(msg) {
+      if (msg.type === 'new') {
+        $scope.currentActiveDeploys += 1;
+      } else if (msg.type === 'finish') {
+        $scope.currentActiveDeploys -= 1;
+      }
+      updateBadge();
+    });
+  }, 5000);
 
   function updateBadge() {
     _.defer(function() { $scope.$apply(); });
     if ($scope.currentActiveDeploys > 0) {
-      $('#currentDeploysBadge').show();
+      $badge.show();
     } else {
       $scope.currentActiveDeploys = 0;
-      $('#currentDeploysBadge').hide();
+      $badge.hide();
     }
   }
 
-  function init() {
-    $http.get('/deploys/active_count.json').success(function(result) {
```

Comment:
Yeah :/

What do you think about getting rid of the live counter ?

On Mon, May 16, 2016 at 1:36 AM, Shane Hender <notifications@github.com>
wrote:

> In app/assets/javascripts/controllers/current_deploys_badge_ctrl.js
> <https://github.com/zendesk/samson/pull/985#discussion_r63323543>:
>
> >      }
> >    }
> >
> > -  function init() {
> > -    $http.get('/deploys/active_count.json').success(function(result) {
>
> You will still need this call otherwise the badge count will be
> potentially incorrect since it'll miss all the starting/stopping deploy
> events during the 5 seconds.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/985/files/4ff90ba109829c24707de930034934a3cf56ba80#r63323543>
>